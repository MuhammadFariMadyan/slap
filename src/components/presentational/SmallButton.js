import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Button = styled.button`
  background: #eb4d4b;
  color: white;
  border: 0px;
  padding: 4px 8px;
  font-size: 11px;
  outline: none;
  cursor: pointer;

  &:hover {
    background: #ce3d3b;
  }
`;

const SmallButton = ({ caption, onClick }) => (
  <Button onClick={onClick}>{caption}</Button>
);

SmallButton.propTypes = {
  caption: PropTypes.string,
  onClick: PropTypes.func
};

export default SmallButton;
