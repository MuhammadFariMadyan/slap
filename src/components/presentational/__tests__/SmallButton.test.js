import React from "react";
import { shallow } from "enzyme";
import "jest-styled-components";

import SmallButton from "../SmallButton";

describe("Thumbnail", () => {
  const setup = propOverrides => {
    const props = {
      caption: "My Own Button",
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<SmallButton {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClick' callback while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });
});
