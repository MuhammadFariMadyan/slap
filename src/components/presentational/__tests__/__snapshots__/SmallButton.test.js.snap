// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail should renders default 1`] = `
<styled.button
  onClick={[Function]}
>
  My Own Button
</styled.button>
`;
